homeassistant/components/evohome/__init__.py

```diff
@@ -200,9 +199,16 @@ async def load_auth_tokens(store: Store) -> tuple[dict, dict | None]:
     hass.data[DOMAIN]["broker"] = broker = EvoBroker(
         hass, client_v2, client_v1, store, config[DOMAIN]
     )
-
     await broker.save_auth_tokens()
-    await broker.async_update()  # get initial state
+
+    hass.data[DOMAIN]["coordinator"] = coordinator = EvoCoordinator(
+        hass,
+        _LOGGER,
+        name=f"{DOMAIN}_coordinator",
+        update_interval=config[DOMAIN][CONF_SCAN_INTERVAL],
+        update_method=broker.async_update,
+    )
+    await coordinator.async_refresh()  # get initial state now
```

Comment:
async_initial_refresh?

Reply:
`async_refresh` is right (unless I am missing something?).

I am not subclassing this method from **DataUpdateCoordinator**.

Reply:
When called from setup we call a different function that allow exceptions for setup retry.

Reply:
More specifically `coordinator.async_config_entry_first_refresh()`

https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities

Reply:
How can I do retry without config flow? I have had a look through the other integrations for an example, but such examples - if any - would be rare, and I couldn't find any.

Reply:
Hmmm... How about I subclass `_async_refresh`? Lemmee see...

Reply:
Stop.. just call that method instead. Remove the subclass and add the listener in async_setup() then you are done.